pkg/service/catalog.go

```diff
@@ -157,160 +145,113 @@ func (c *catalog) GetService(serviceID string) (Service, bool) {
 
 // NewService initialized and returns a new Service
 func NewService(
-	serviceProperties *ServiceProperties,
+	serviceProperties ServiceProperties,
 	serviceManager ServiceManager,
 	plans ...Plan,
 ) Service {
-	s := &service{
+	s := service{
 		ServiceProperties: serviceProperties,
 		serviceManager:    serviceManager,
 		plans:             plans,
 		indexedPlans:      make(map[string]Plan),
 	}
 	for _, planIfc := range s.plans {
-		p := planIfc.(*plan)
-		p.Schemas.addCommonSchema(serviceProperties)
+		p := planIfc.(plan)
```

Comment:
Shouldn't have to make this type assertion here. Just add `planIfc` to the indexed plans instead. Better yet, just change `for _, planIfc := range s.plans {` to `for _, p := range s.plans {`.